demos/exyielding/README.md

```diff
@@ -0,0 +1,27 @@
+# Exyielding
+
+In this demo we want to show that a thread can yield the utilization of the CPU directly to another thread.
+
+To illustrate the mechanism in it, we will explain:
+
+* `yield_to` (in `src/user/thread.c`)
+
+## `yield_to`
+
+When this function is called, the kernel will check if the thread we specified is available to yield to.
+
+If the thread can be yielded to, we will set `next_thread` to the address of this thread and call the `do_scheduler` function.
+
+Since we have decided `next_thread`, the scheduler will do housekeeping tasks and help us to setup the thread we had already chosen.
+
+If the thread we specified is unavailable to yield, `do_scheduler` will not be called and we will go back to the point we called `yield_to()`.
+
+## Walkthrough
+
+At the beginning of this demo, we create two named thread `first` and `second`. After the thread `setup` ends, the thread `first` will get the use of the CPU and yield to the thread `second` specified by its thread ID (`tid`).
+
+The thread `second` will first yield to the thread `first`, then the thread `first` will end. When thread `second` is yielded to again, it creates a new thread `last`. This thread is used to show that we can yield to higher `tid`  as well as the lower one.
```

Comment:
(just gonna quote you here for convenience)

> I am thinking that maybe we can create the thread last first then do yielding to the thread first. It will be more clear to show that we can yield to the higher or the lower tid as well. Would it be better?

I remembered the original reason the place we add the "last" thread.
It was so that we could `yield_next` and go from:
```
[setup (exited)][first][second (current)][(invalid)]
```
to:
```
[setup (exited)][first (current)][second][(invalid)]
```
Without having `last` on the end there to get in the way.

Anyway.

> Would it be better?

Absolutely. I would add all the threads in setup() and also rename the first/second worker functions to first() and second() just to be a bit clearer. Make one of the yields here:
```
  yield(); // switch to "last"
  yield(); // switch back to "last"
```
A yield _to.

Reply:
In general this demo has gone from showing `yield_next` and `yield_to` to just `yield_to` so any change to focus on the latter is fine.

Reply:
Shall we work on changing this demo first?

Reply:
Sure, do it as the first commit of this PR.